crates/trie/trie/src/proof_v2/mod.rs

```diff
@@ -570,27 +653,456 @@ where
         }
     }
 
+    /// Given the lower and upper bounds (exclusive) of a range of keys, iterates over the
+    /// `hashed_cursor` and calculates all trie nodes possible based on those keys. If the upper
+    /// bound is None then it is considered unbounded.
+    ///
+    /// It is expected that this method is "driven" by `next_uncached_key_range`, which decides
+    /// which ranges of keys need to be calculated based on what cached trie data is available.
+    #[instrument(
+        target = TRACE_TARGET,
+        level = "trace",
+        skip(self, value_encoder, targets, hashed_cursor_current),
+    )]
+    fn calculate_key_range(
```

Comment:
This method now contains the logic which was previously in `proof_inner` to construct the trie based on a sequence of keys (leafs). It's only significant change is that it is now bounded.